gplugins/lumerical/tests/test_lumerical_utils.py

```diff
@@ -0,0 +1,249 @@
+
+
+def test_to_lbr():
+    # Inputs
+    layer_map = {
+        "si": "Si (Silicon) - Palik",
+        "sio2": "SiO2 (Glass) - Palik",
+        "sin": "Si3N4 (Silicon Nitride) - Phillip",
+        "TiN": "TiN - Palik",
+        "Aluminum": "Al (Aluminium) Palik",
+    }
+
+    # Create layerstack
+    layerstack_lumerical2021 = LayerStack(
+        layers={
+            "clad": LayerLevel(
+                name=None,
+                layer=(99999, 0),
+                thickness=3.0,
+                thickness_tolerance=None,
+                zmin=0.0,
+                zmin_tolerance=None,
+                material="sio2",
+                sidewall_angle=0.0,
+                sidewall_angle_tolerance=None,
+                width_to_z=0.0,
+                z_to_bias=None,
+                mesh_order=9,
+                layer_type="background",
+                mode=None,
+                into=None,
+                resistivity=None,
+                bias=None,
+                derived_layer=None,
+                info={},
+                background_doping_concentration=None,
+                background_doping_ion=None,
+                orientation="100",
+            ),
+            "box": LayerLevel(
+                name=None,
+                layer=(99999, 0),
+                thickness=3.0,
+                thickness_tolerance=None,
+                zmin=-3.0,
+                zmin_tolerance=None,
+                material="sio2",
+                sidewall_angle=0.0,
+                sidewall_angle_tolerance=None,
+                width_to_z=0.0,
+                z_to_bias=None,
+                mesh_order=9,
+                layer_type="background",
+                mode=None,
+                into=None,
+                resistivity=None,
+                bias=None,
+                derived_layer=None,
+                info={},
+                background_doping_concentration=None,
+                background_doping_ion=None,
+                orientation="100",
+            ),
+            "core": LayerLevel(
+                name=None,
+                layer=(1, 0),
+                thickness=0.22,
+                thickness_tolerance=None,
+                zmin=0.0,
+                zmin_tolerance=None,
+                material="si",
+                sidewall_angle=10.0,
+                sidewall_angle_tolerance=None,
+                width_to_z=0.5,
+                z_to_bias=None,
+                mesh_order=2,
+                layer_type="grow",
+                mode=None,
+                into=None,
+                resistivity=None,
+                bias=None,
+                derived_layer=None,
+                info={"active": True},
+                background_doping_concentration=100000000000000.0,
+                background_doping_ion="Boron",
+                orientation="100",
+            ),
+            # KNOWN ISSUE: Lumerical 2021 version of Layer Builder does not support dopants in process file
```

Comment:
is there a way we can check the lumerical version (or try/catch somewhere)? being able to draw dopant layers is a pretty important functionality (and the dopant drawings have some extra logic which would benefit from having test coverage)

Reply:
@elamdf We can get the application version (https://optics.ansys.com/hc/en-us/articles/360034932053-version-Script-command), but it does not tell us whether it is from 2021 and prior. The format is like this "7.28.3578". I attempted to look through Lumerical's object API but have yet to find something that may indicate the year in which the software is released. Attempted to look for release notes with these version numbers, but it's not readily apparent...

Reply:
The following error message is from Lumerical:
lumapi.LumApiError: 'Please import your process file using the layer builder editor window for more options.'

It's sufficient to warn users of an issue but not specific enough to root cause. Good idea on adding try/catch. I'm going to rework the geometry import to include a better error message.

Reply:
https://github.com/seanlam97/gplugins/commit/a9b62d82b43bd3d69df6434ea5b3799143a06a24